tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -921,9 +923,19 @@ static void test_cancelled_alarm_does_not_expire_instance(const struct device *d
 
 		cnt = IS_ENABLED(CONFIG_ZERO_LATENCY_IRQS) ?
 			alarm_cnt : k_sem_count_get(&alarm_cnt_sem);
-		zassert_equal(0, cnt,
+		/* For slower processors it may happen that alarm expires before
+		 * it is canceled. That is accepted scenario and it is detected
+		 * by checking that timestamp in the last alarm handler was for
+		 * the current alarm.
+		 */
+		zassert_true((cnt == 0) || (last_alarm_handler >= alarm_cfg.ticks),
```

Comment:
Unrelated changes? Move to another PR?

Reply:
Ups, Started from wrong branch. Will remove.